src/main/common/filter.h

```diff
@@ -40,6 +41,14 @@ typedef struct biquadFilter_s {
     float x1, x2, y1, y2;
 } biquadFilter_t;
 
+typedef struct laggedMovingAverage_s {
+    int movingWindowIndex;
+    int windowSize;
```

Comment:
This doesn't match the `uint8_t` in `laggedMovingAverageInit()`.

Reply:
Yeah, we probably shouldn't limit to uint8_t.  It's not like we can ever make a huge buffer but > 255 might be reasonable.

Reply:
fixed

Reply:
`windowSize` should definitely not be `int`, setting it to a negative value will make this break. I'd recommend to go `uint16_t`, having more than 128kB buffer (and 1s lag time at 32kHz) doesn't sound very realistic.

Reply:
none of them should be signed.. I'll set them to uint16